xapian-core/weight/tfidfweight.cc

```diff
@@ -197,11 +180,12 @@ TfIdfWeight::unserialise(const string & s) const
     const char *end = ptr + s.size();
     double slope = unserialise_double(&ptr, end);
     double delta = unserialise_double(&ptr, end);
+    if (rare(end - ptr != 3))
+	throw Xapian::SerialisationError
+	      ("Extra data in TfIdfWeight::unserialise()");
```

Comment:
OK, except that it might be more or less data now - "Incorrect data ..." maybe?

Reply:
Updated